packages/frontend/src/components/MetricQueryData/UnderlyingDataModal.tsx

```diff
@@ -1,18 +1,35 @@
 import { Dialog } from '@blueprintjs/core';
 import React, { FC } from 'react';
+import { useTracking } from '../../providers/TrackingProvider';
+import { EventName } from '../../types/Events';
 import { useMetricQueryDataContext } from './MetricQueryDataProvider';
 import UnderlyingDataModalContent from './UnderlyingDataModalContent';
 
-interface Props {}
-
-const UnderlyingDataModal: FC<Props> = () => {
+const UnderlyingDataModal: FC<{
+    trackingData: {
+        organizationId: string | undefined;
+        userId: string | undefined;
+        projectId: string | undefined;
+    };
+}> = ({ trackingData }) => {
     const { isUnderlyingDataModalOpen, closeUnderlyingDataModal } =
         useMetricQueryDataContext();
+    const { track } = useTracking();
 
     return (
         <Dialog
             isOpen={isUnderlyingDataModalOpen}
-            onClose={closeUnderlyingDataModal}
+            onClose={() => {
+                closeUnderlyingDataModal();
+                track({
+                    name: EventName.VIEW_UNDERLYING_DATA_CLICKED,
```

Comment:
I don't think this event should be triggered with the modal is closed. Otherwise how can we distinguish between opening and closing underlying data modal ?